synapse/replication/http/membership.py

```diff
@@ -85,10 +85,6 @@ async def _handle_request(  # type: ignore
         remote_room_hosts = content["remote_room_hosts"]
         event_content = content["content"]
 
-        requester = Requester.deserialize(self.store, content["requester"])
-
-        request.requester = requester
```

Comment:
Here (and below) the `request` is never used after this line, so I don't think this is doing anything.

Reply:
How confident are we? If we're getting `request` as an input parameter and than mutating it as a side effect, might that have issues elsewhere?

Reply:
I'm not sure what you're asking. Are you asking if we're confident in this change or if the code before this change might have been masking other bugs or something?

Note that this is the replication API, which I don't think has any concept of requester? The `Requester.deserialize` doesn't do much except generate an object from a string, so it shouldn't have any real side-effects. Maybe this is here to change logging?

@erikjohnston do you know what the proper thing to do here is?

Reply:
Yeah, I think is so that the requester is logged, rather than the default `{None}`. We should probably make it happen automatically, somehow, rather than having it only work for a subset of the replication APIs.

Reply:
Ah, OK. The weird thing is that that implies this is actually a `SynapseRequest` then, not a `Request`. I'll see if I can rework this bit.